ocaml-lsp-server/src/inlay_hints.ml

```diff
@@ -56,6 +56,8 @@ let hint_binding_iter ?(hint_let_bindings = false)
       | Texp_match (expr, cases, _) ->
         iter.expr iter expr;
         List.iter cases ~f:(case hint_pattern_variables iter)
+      (* Stop iterating when we see a ghost location to avoid annotating generated code *)
+      | _ when e.exp_loc.loc_ghost -> ()
```

Comment:
Shouldn't the case be above the Tstr_match and continue iteration with the default iterator?

Reply:
What makes you say that? Continuing with the default iterator will create inlay hints on the generated code.

Reply:
That seems to be what's happening. When I build within Jane Street, the tests work fine (presumably because the document is created with non-ghost locations), but building with external libraries causes this issue where the test document has ghost locations. I'm trying to track down the source of the difference.